### ISSUE TITLE
[Question] Writer message separator?

### ISSUE THREAD

**Issue by manugarri** [linkedin/gobblin#1396]
Hi , just wanted to know if there is a parameter for any hdfs writer (I am using *SimpleDataWriterBuilder*) to add a data separator between data points, for example, a new line.

I have a kafka topic with json encoded data and , even though SimpleDataWriterBuilder can write to hdfs in plain text format, it writes every record without a single separation, making it hard to parse in spark (*spark.read.json expects one record per line*).

Im not sure if this already exists (seems pretty common) or if not, if it would make sense to add it as a feature request.

Thanks for this awesome project!

**Comment by ibuenros** [comment]
Yes, there is one:
simple.writer.delimiter

**Comment by manugarri** [comment]
Oh great, im going to try it right now! If i may ask, is there any list of all the possible parameters? How can one learn them besides going deep into the java code?

**Comment by ibuenros** [comment]
Unfortunately different writers (or sources) have different levels of
documentation. The Simple writer in particular is poorly documented. We
have plans on how to improve this, but we're focusing on other features at
the moment, so the documentation will depend on the specific construct
developer for now.

Don't know if you've already seen the official docs, but just in case:
https://gobblin.readthedocs.io/en/latest/

Issac

**Comment by manugarri** [comment]
Yeah, I read the docs. They are very good. I guess its a bit challenging keeping all the dsl documented. 

Anyways, you definitely helped me a lot. Thanks so much!
